tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -322,10 +334,30 @@ def print_(*objects, **kwargs):
   else:
     _py_print(*objects, **kwargs)
 
-
 def _py_print(*objects, **kwargs):
   print(*objects, **kwargs)
 
+def max_(*args, **kwargs):
+  if any(tensor_util.is_tf_type(s) for s in args):
+    return _tf_max(*args, **kwargs)
+  return _py_max(*args, **kwargs)
+
+def _tf_max(*args, **kwargs):
+  if len(kwargs):
+    kwargs_tuple = tuple(set(kwargs.keys()))
+    raise ValueError('These keyword arguments are ' 
+                     'currently not supported: {}'.format(kwargs_tuple))
+  elif len(args) == 1:
+    # TODO (bhack) Do we want the input Tensor to be rank==1 or scalar?
```

Comment:
I think we need to have exactly one element, ie. `tf.size(t) == 1`.

Reply:
Are you sure you want to limit to 1? I think that here we could have

```python
import tensorflow as tf
@tf.function
def test(*args):
  if len(args)==1:
    return tf.reduce_max(*args, axis=0)
  else:
    return -1
assert test(tf.constant([5, 6, 7, 8])).numpy() == max([5, 6, 7, 8])
```

Reply:
Ah, I was looking in the wrong place. You're right, we can take any size as long as the any other dimensions are 1. So [AnyDim], [AnyDim, 1], [AnyDim, 1, 1] would all be ok. (At least that's what numpy does).

Reply:
Yes but I don't think that we want to map numpy here right? I think. we just want to map a subset of python `max`.

So do you think that we need to limit this failing case?
```python
assert test(tf.constant([[5, 6, 7, 8][)).numpy() == max([5, 6, 7, 8])

----> 8 assert test(tf.constant([[5, 6, 7, 8]])).numpy() == max([5, 6, 7, 8])
```

Reply:
Oh no, I was just using numpy as example. We definitely don't want to map here.

At any rate, I think we can start simple - just check that the rank is 1. We can always add more cases later.

Reply:
Yes if you see it is exactly the comment at line n. 351. I suppose that we need to duplicate your code gist below for this?

Reply:
Technically, we would, but I'm also thinking that we could start simple (and also below). We could just require `shape.rank == 1` for starters. That would reject dynamic rank cases, or dynamic shapes, but those are less frequent. So things like `tf.constant([1, 2, 3])` and `tf.range(3)` would work, but `tf.range(tf.random.uniform(3))` would not.

Reply:
So do you think that this could be enough or we need to care about the "magic" in the context manger?
```python
  if len(args)==1:
    rank = tf.rank(*args)
    if rank == 1:
      return tf.reduce_max(*args, axis=0)
    else:
       ValueError...
```